src/tests/QueryBuilderSpec.scala

```diff
@@ -21,6 +23,31 @@ case class QueryBuilderSpec()(implicit runner: Runner) {
       .build()
       .getService
   }
+  
+  implicit val idx = Schema.load(
+    DatastoreIndex[QueringTestEntity](
+      ("intParam", OrderDirection.Descending),
+      ("innerClass.intParam", OrderDirection.Ascending)
+    ),
+    DatastoreIndex[QueringTestEntity](
+      ("intParam", OrderDirection.Ascending),
+      ("innerClass.intParam", OrderDirection.Descending),
+      ("innerClass.decimalParam", OrderDirection.Ascending)
+    )
+  )
+
+  Dao[QueringTestEntity].all
+    .filter(_.intParam == 0)
+    .filter(_.innerClass.intParam == 1)
+    .filter(_.innerClass.intParam < 102)
+    .filter(_.innerClass.intParam > 2)
+    // .filter(_.intParam >= 1) // It would not compile if uncommented, due to query validations. It's correct behavior as there can be only 1 property with inequality filter
+    .sortBy(_.intParam)
+    .reverse
+    .sortBy(_.innerClass.intParam)
+    .reverse  // Would result in `order by innerClass.intParam DESC, intParam ASC`
+    // .sortBy(_.innerClass.decimalParam) //If uncommented this line should not compile, since inequality proporty is not first (or last in tems of this DSL) sort order criteria
+    .run()
```

Comment:
```
val schema = Schema(<definitions>) // just defines the schema, returns an complex type
// check the schema against the remote database, and use an implicit proof of the schema to execute queries.
schema.using { implicit schema => // this `schema` value would have a different type to `Schema`, because it's *proof* that the database has the schema
// run database queries 
// return some T
} // returns a Result[T]
```